v-next/hardhat/src/internal/builtin-plugins/coverage/coverage-manager.ts

```diff
@@ -2,49 +2,54 @@ import type {
   CoverageData,
   CoverageManager,
   CoverageMetadata,
-  Statement,
-  Tag,
 } from "./types.js";
 import type { TableItem } from "@nomicfoundation/hardhat-utils/format";
 
 import path from "node:path";
 
-import { assertHardhatInvariant } from "@nomicfoundation/hardhat-errors";
 import { divider, formatTable } from "@nomicfoundation/hardhat-utils/format";
 import {
   ensureDir,
   getAllFilesMatching,
   readJsonFile,
+  readUtf8File,
   remove,
   writeJsonFile,
   writeUtf8File,
 } from "@nomicfoundation/hardhat-utils/fs";
 import chalk from "chalk";
 import debug from "debug";
 
+import { getProcessedCoverageInfo } from "./process-coverage.js";
+
 const log = debug("hardhat:core:coverage:coverage-manager");
 
 const MAX_COLUMN_WIDTH = 80;
 
 type Line = number;
-type Branch = [Line, Tag];
 
 /**
  * @private exposed for testing purposes only
  */
+export interface FileReport {
+  // NOTE: currently, the counters for how many times a statement is executed are not implemented in EDR,
+  // so the only information available is whether a statement was executed, not how many times it was executed.
+  // Also, branch coverage is not available.
+  // In addition, partially executed lines (for example, ternary operators) cannot be determined, as this information is missing in EDR,
+  // since only whole lines can be registered as executed or not.
```

Comment:
```suggestion
  // since only whole statements can be registered as executed or not.
```

Suggestion: EDR implements instrumentation for statement-coverage. If a line contains multiple statements, EDR would be able to report it